plugins/gradle/src/org/jetbrains/plugins/gradle/GradleManager.java

```diff
@@ -144,7 +145,11 @@ public Function<Pair<Project, String>, GradleExecutionSettings> getExecutionSett
       if (!StringUtil.isEmpty(javaHome)) {
         LOG.info("Instructing gradle to use java from " + javaHome);
       }
-      result.setJavaHome(javaHome);
+
+      if (projectLevelSettings == null || !GradleDaemonJvmHelper.isProjectUsingDaemonJvmCriteria(projectLevelSettings)) {
```

Comment:
Could you also cover the previous code block with this if-block? We don't need to resolve the `javaHome` path if it won't be used in result execution settings.

```
if (projectLevelSettings == null || !GradleDaemonJvmHelper.isProjectUsingDaemonJvmCriteria(projectLevelSettings)) {
  String javaHome = gradleInstallationManager.getGradleJvmPath(project, rootProjectPath);
  if (!StringUtil.isEmpty(javaHome)) {
    LOG.info("Instructing gradle to use java from " + javaHome);
  }
  result.setJavaHome(javaHome);
}
```